jwt/jwt.go

```diff
@@ -118,6 +118,44 @@ func getJWK(ctx context.Context, kid string) (*clerk.JSONWebKey, error) {
 	return nil, fmt.Errorf("no jwk key found for kid %s", kid)
 }
 
+// Returns the JSON web key set. Tries a cached value first, but if
+// there's no value or the entry has expired, it will fetch the set
+// from the API and cache the value.
+func getJWKSWithCache(ctx context.Context) (*clerk.JSONWebKeySet, error) {
+	cacheKey := "/v1/jwks"
+	var jwks *clerk.JSONWebKeySet
+	var err error
+
+	// Try the cache first. Make sure we have a non-expired entry and
+	// that the value is a valid JWKS.
+	entry, ok := getCache().Get(cacheKey)
```

Comment:
🔧 I think the cache must have the KID (instance ID) as key (or part of it). You can then quickly lookup by KID since that's the requirement from line 114. This way you can also have a multi-instance scenario handled.

Reply:
Good thinking. I'm just a bit reluctant to change the existing behavior. :thinking: 

As far as I understand the cache stores the whole json web key set response from the server. It was like this before and I'm a bit hesitant to change it now.

Regarding multi-instance scenarios, indeed, caching per instance is not possible with this setup, but I don't think it's needed for now. 

I'll toy a bit with your proposal and open a separate PR if it doesn't break anything. Assuming that's alright, since you added a :wrench: comment. :smile: